engine/overworld/weather.asm

```diff
@@ -0,0 +1,778 @@
+DoOverworldWeather:
+	push de
+	push hl
```

Comment:
The order is always to push `af` `hl`, `de`, `bc` and pop in reverse, unless there's an optimization reason not to.
```suggestion
	push hl
	push de
```